src/sql/workbench/services/query/common/query.ts

```diff
@@ -75,3 +80,11 @@ export interface ICellValue {
 	isNull?: boolean;
 	invariantCultureDisplayValue?: string;
 }
+
+export interface IQueryResultsWriter {
+	onQueryStart(): void | Promise<void>;
```

Comment:
Is there a reason that a result writer needs to expose these vs just passing in the running when constructing the writer instances and letting them handle hooking up the handlers? (you could do that in a base class common to all result writers to avoid duplicating code)